src/main/java/fr/mossroy/tomcat/tools/ExternalPropertySource.java

```diff
@@ -31,8 +33,9 @@
     private static final String CATALINA_PROPERTIES = "conf/catalina.properties";
     private static final String CATALINA_PROPERTIES_FILE_PROPERTY = "fr.mossroy.tomcat.tools.ExternalPropertySource.file";
     private static final Log LOGGER = LogFactory.getLog(ExternalPropertySource.class);
+    private static final Pattern regExp = Pattern.compile("\\$\\{([^\\}]*)\\}");

```

Comment:
It would probably be better to not match `${}` (with no property name inside the brackets)
Using `\\$\\{([^\\}]+)\\}` instead of `\\$\\{([^\\}]*)\\}` should be enough